kedro/config/omegaconf_config.py

```diff
@@ -109,6 +111,19 @@ def __init__(
         # It's easier to introduce them step by step, but removing them would be a breaking change.
         self._clear_omegaconf_resolvers()
 
+        file_mimetype, _ = mimetypes.guess_type(conf_source)
```

Comment:
I think that's because when you run `tar -czf tar_conf conf` you don't actually seem to create a valid tar file. At least on my machine it's not shown as a compressed file and I also can't open it as such. I'll clarify in the docs what commands people should run both for valid `tar.gz` and `zip` files.